core/templates/dev/head/domain/story_viewer/ReadOnlyStoryNodeObjectFactory.ts

```diff
@@ -17,72 +17,92 @@
  * story viewer.
  */
 
-angular.module('oppia').factory('ReadOnlyStoryNodeObjectFactory', [function() {
-  var StoryNode = function(
-      id, title, destinationNodeIds, prerequisiteSkillIds, acquiredSkillIds,
-      outline, outlineIsFinalized, explorationId, explorationSummary,
-      completed) {
-    this._id = id;
-    this._title = title;
-    this._destinationNodeIds = destinationNodeIds;
-    this._prerequisiteSkillIds = prerequisiteSkillIds;
-    this._acquiredSkillIds = acquiredSkillIds;
-    this._outline = outline;
-    this._outlineIsFinalized = outlineIsFinalized;
-    this._explorationId = explorationId;
-    this._explorationSummaryObject = explorationSummary;
-    this._completed = completed;
-  };
+import { downgradeInjectable } from '@angular/upgrade/static';
+import { Injectable } from '@angular/core';
 
-  // Instance methods
+export interface ExplorationSummary {
+  title: string,
+  status: string
+}
 
-  StoryNode.prototype.getId = function() {
-    return this._id;
-  };
+export class StoryNode {
```

Comment:
Since we now have a `ReadOnlySubtopicPage` object, rename this to `ReadOnlyStoryNode` for consistency.

Reply:
Done